### PR TITLE
[Misc] Add `ti.core.toggle_advanced_optimization(True/False)`

### PR DESCRIPTION
**Pull request by yuanming-hu**
Related issue = #926 

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

File: taichi/python/export_lang.cpp [hunk 1]
```diff
@@ -537,6 +537,10 @@ void export_lang(py::module &m) {
   m.def("is_supported", is_supported);
 
   m.def("print_stat", [] { stat.print(); });
+
+  // A temporary option which will be removed soon in the future
+  m.def("toggle_advanced_optimization",
+        [](bool option) { advanced_optimization = option; });
```
Comment (archibate) [inline taichi/python/export_lang.cpp:543]: How about add `$TI_ADVANCED_OPTIMIZATION` too?

File: taichi/program/compile_config.cpp [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 TLANG_NAMESPACE_BEGIN
 
-const bool advanced_optimization = true;
+bool advanced_optimization = true;
```
Comment (archibate) [inline taichi/program/compile_config.cpp:5]: Why not move this inside the `ti.cfg`?

Comment (yuanming-hu) [reply taichi/program/compile_config.cpp:5]: This is just a temporary solution to disable advanced optimization. Moving this to `ti.cfg` needs too many changes.

File: taichi/python/export_lang.cpp (hunk 1 above)
Comment (yuanming-hu) [reply taichi/python/export_lang.cpp:543]: This feature is just for a small group of users in special needs, so I'd rather not make it too flexible...

**Pull request merged**